src/lib/mailer/index.js

```diff
@@ -101,11 +101,18 @@ export const onFailedNotification = (job, error) => {
 };
 
 export const onCompletedNotification = (job, result) => {
-  if (result != null) {
-    logger.info(`Successfully sent ${job.name} notification for ${job.data.report.displayId || job.data.report.id}`);
-    logEmailNotification(job, true, result);
+  if (job.data.reports && Array.isArray(job.data.reports)) {
+    job.data.reports.forEach((report) => {
+      if (result != null) {
+        logger.info(`Successfully sent ${job.name} notification for ${report.displayId}`);
+      } else {
+        logger.info(`Did not send ${job.name} notification for ${report.displayId} preferences are not set or marked as "no-send"`);
+      }
+    });
+  } else if (result != null) {
```

Comment:
Again is there any concern for accessing null values with things like job.data.report.displayId?